test/QUICKSTART-KIND.md

```diff
@@ -0,0 +1,305 @@
+# 1. Create kind cluster
+kind create cluster --name gabi-integration
+
+# 2. Build and load test image
+make integration-test-image
+kind load docker-image gabi-integration-test:local --name gabi-integration
+
+# 3. Pull and load supporting service images
+podman pull registry.redhat.io/rhel9/postgresql-16:9.6
+podman pull quay.io/app-sre/splunk:latest
+kind load docker-image registry.redhat.io/rhel9/postgresql-16:9.6 --name gabi-integration
+kind load docker-image quay.io/app-sre/splunk:latest --name gabi-integration
+
+# 4. Deploy supporting services
+kubectl apply -f test/test-pod.yml
+kubectl wait --for=condition=ready pod/test-pod --timeout=300s
+
+# 5. Deploy test runner (create your own YAML or use the script's template)
+kubectl run gabi-test \
+  --image=gabi-integration-test:local \
+  --image-pull-policy=Never \
+  --restart=Never \
+  --env="DB_HOST=test-pod" \
+  --env="DB_PORT=5432" \
+  --env="DB_USER=gabi" \
+  --env="DB_PASS=passwd" \
+  --env="DB_NAME=mydb" \
+  --env="SPLUNK_ENDPOINT=http://test-pod:8088"
+
+# 6. Watch logs
+kubectl logs -f gabi-test
+
+# 7. Check exit code
+kubectl get pod gabi-test -o jsonpath='{.status.containerStatuses[0].state.terminated.exitCode}'
+
+# 8. Cleanup
+kubectl delete pod/gabi-test pod/test-pod
+kind delete cluster --name gabi-integration
+```
+
+## Understanding the Test Image
+
+The `test/Dockerfile.integration` creates an image with:
+
+```
+/usr/local/bin/
+  ├── gabi              # Main application binary
+  └── integration.test  # Compiled test binary
+
+Debugging tools included:
+  ├── psql              # PostgreSQL client
+  ├── nc                # Network connectivity testing (ncat)
+  ├── nslookup/dig      # DNS resolution testing
+  └── curl              # HTTP testing
+```
+
+Run tests with:
+```bash
+podman run --rm gabi-integration-test:local
+```
+
+Run just the gabi app:
+```bash
+podman run --rm gabi-integration-test:local /usr/local/bin/gabi
+```
+
+Debug connectivity:
+```bash
+# Test database connection
+podman run --rm gabi-integration-test:local psql -h test-pod -U gabi -d mydb -c "SELECT 1"
+
+# Test network connectivity
+podman run --rm gabi-integration-test:local nc -zv test-pod 5432
+```
+
+## How Tests Work
+
+### ✅ Tests Are Now Cluster-Ready!
+
+The integration tests in `integration_test.go` have been **updated** to work with deployed services:
+
+- ✅ No longer need **gnomock/testcontainers** in Kubernetes
+- ✅ Connect to services via **environment variables**
+- ✅ Work in **Kubernetes pods** without special privileges
+- ✅ Can still run **locally** with services on localhost
+
+### Example
+
+```go
+func TestHealthCheckOK(t *testing.T) {
+    // Reads from environment or uses defaults
+    dbHost := getEnvOrDefault("DB_HOST", "localhost")
+    dbPort := getEnvOrDefault("DB_PORT", "5432")
+
+    // Connects to deployed services
+    setEnvironment(configFile, dbHost, dbPort, ...)
+    // ... test logic
+}
+```
+
+For more details, see `test/INTEGRATION-SETUP.md`
```

Comment:
not interested in this here, the documentation of integration tests should cover this.